src/configuration-requestor.spec.ts

```diff
@@ -111,97 +115,217 @@ describe('ConfigurationRequestor', () => {
-        const responseFile = url.includes('bandits')
-          ? MOCK_BANDIT_MODELS_RESPONSE_FILE
-          : MOCK_FLAGS_WITH_BANDITS_RESPONSE_FILE;
-        const response = readMockUFCResponse(responseFile);
-
+        const response = responseMockGenerator(url);
         return Promise.resolve({
           ok: true,
           status: 200,
           json: () => Promise.resolve(response),
         });
       }) as jest.Mock;
       global.fetch = fetchSpy;
-    });
+    }
 
-    it('Fetches and populates bandit parameters', async () => {
-      await configurationRequestor.fetchAndStoreConfigurations();
+    function responseMockGenerator(url: string) {
+      const responseFile = url.includes('bandits')
+        ? MOCK_BANDIT_MODELS_RESPONSE_FILE
+        : MOCK_FLAGS_WITH_BANDITS_RESPONSE_FILE;
+      return readMockUFCResponse(responseFile);
+    }
 
-      expect(fetchSpy).toHaveBeenCalledTimes(2); // Once for UFC, another for bandits
-
-      expect(flagStore.getKeys().length).toBeGreaterThanOrEqual(2);
-      expect(flagStore.get('banner_bandit_flag')).toBeDefined();
-      expect(flagStore.get('cold_start_bandit')).toBeDefined();
-
-      expect(banditModelStore.getKeys().length).toBeGreaterThanOrEqual(2);
-
-      const bannerBandit = banditModelStore.get('banner_bandit');
-      expect(bannerBandit?.banditKey).toBe('banner_bandit');
-      expect(bannerBandit?.modelName).toBe('falcon');
-      expect(bannerBandit?.modelVersion).toBe('v123');
-      const bannerModelData = bannerBandit?.modelData;
-      expect(bannerModelData?.gamma).toBe(1);
-      expect(bannerModelData?.defaultActionScore).toBe(0);
-      expect(bannerModelData?.actionProbabilityFloor).toBe(0);
-      const bannerCoefficients = bannerModelData?.coefficients || {};
-      expect(Object.keys(bannerCoefficients).length).toBe(2);
-
-      // Deep dive for the nike action
-      const nikeCoefficients = bannerCoefficients['nike'];
-      expect(nikeCoefficients.actionKey).toBe('nike');
-      expect(nikeCoefficients.intercept).toBe(1);
-      expect(nikeCoefficients.actionNumericCoefficients).toHaveLength(1);
-      const nikeBrandAffinityCoefficient = nikeCoefficients.actionNumericCoefficients[0];
-      expect(nikeBrandAffinityCoefficient.attributeKey).toBe('brand_affinity');
-      expect(nikeBrandAffinityCoefficient.coefficient).toBe(1);
-      expect(nikeBrandAffinityCoefficient.missingValueCoefficient).toBe(-0.1);
-      expect(nikeCoefficients.actionCategoricalCoefficients).toHaveLength(2);
-      const nikeLoyaltyTierCoefficient = nikeCoefficients.actionCategoricalCoefficients[0];
-      expect(nikeLoyaltyTierCoefficient.attributeKey).toBe('loyalty_tier');
-      expect(nikeLoyaltyTierCoefficient.missingValueCoefficient).toBe(0);
-      expect(nikeLoyaltyTierCoefficient.valueCoefficients).toStrictEqual({
-        gold: 4.5,
-        silver: 3.2,
-        bronze: 1.9,
+    describe('Fetching bandits', () => {
+      beforeAll(() => {
```

Comment:
@aarsilv @typotter I'm thinking about moving the polling tests to a isolated test suite and then reset `fetchSpy` to default state  using `afterAll()`

```javascript
describe('ConfigurationRequestor', {
     describe('Bandits with flags', {
        describe('Bandits polling', {
            afterAll(() => {
                initiateFetchSpy(defaultResponseMockGenerator);
            })
            // polling tests here
          });
     });
});
```

Reply:
always a fan of better organization. maybe in a stacked or follow-on PR would be better